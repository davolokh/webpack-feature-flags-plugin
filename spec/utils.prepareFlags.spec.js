const {
  prepareFlags,
} = require('../src/utils');
const sampleFeaturesConfig = require('./webpack.feature-flags.sample.config');


describe('prepareFlags', () => {
  it('prepareFlags: empty config defined', () => {
    expect(prepareFlags(sampleFeaturesConfig, {})).toMatchSnapshot();
  });

  it('prepareFlags: no namespace defined, DEV mode', () => {
    expect(prepareFlags(sampleFeaturesConfig, { mode: 'DEV' })).toMatchSnapshot();
  });

  it('prepareFlags: no namespace defined, PROD mode', () => {
    expect(prepareFlags(sampleFeaturesConfig, { mode: 'PROD' })).toMatchSnapshot();
  });

  it('prepareFlags: TEST namespace defined', () => {
    expect(prepareFlags(sampleFeaturesConfig, { namespace: 'TEST', mode: 'DEV' })).toMatchSnapshot();
  });
});
