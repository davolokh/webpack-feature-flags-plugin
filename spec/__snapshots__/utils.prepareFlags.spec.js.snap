// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prepareFlags prepareFlags: TEST namespace defined 1`] = `
Object {
  "TEST": Object {
    "nameSpace2": Object {
      "feature20": Object {
        "enabled": "false",
      },
      "feature21": Object {
        "enabled": "false",
      },
    },
    "nameSpace3": Object {
      "feature30": Object {
        "enabled": "false",
      },
    },
    "namespace1": Object {
      "feature10": Object {
        "enabled": "true",
      },
      "feature11": Object {
        "enabled": "true",
      },
    },
  },
}
`;

exports[`prepareFlags prepareFlags: empty config defined 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "false",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "false",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;

exports[`prepareFlags prepareFlags: no namespace defined, DEV mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "false",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "true",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;

exports[`prepareFlags prepareFlags: no namespace defined, PROD mode 1`] = `
Object {
  "nameSpace2": Object {
    "feature20": Object {
      "enabled": "true",
    },
    "feature21": Object {
      "enabled": "false",
    },
  },
  "nameSpace3": Object {
    "feature30": Object {
      "enabled": "false",
    },
  },
  "namespace1": Object {
    "feature10": Object {
      "enabled": "false",
    },
    "feature11": Object {
      "enabled": "true",
    },
  },
}
`;
